strawberryfields/ops.py

```diff
@@ -569,15 +569,14 @@ def apply(self, reg, backend, **kwargs):
         """
         values = super().apply(reg, backend, **kwargs)
         # convert the returned values into an iterable with the measured modes indexed along
-        # the first axis and shots along second axis (if larger than 1), so that we can assign
+        # the first axis and shots along second axis, so that we can assign
         # register values
         shots = kwargs.get("shots", 1)
         if self.ns == 1:
             values = [values]  # values is either a scalar, or has shape (shots,)
```

Comment:
This check (and the transpose below) should be unnecessary too, if the backend API funcs always return measurement results in a 2d array with shape == (modes, shots) or shape == (modes, batch_size, shots).